### PR TITLE
Support error output channel internally

### PR DESCRIPTION
**Pull request by t-wissmann**
In the command system, the 'Output' parameter now also carries an error
channel and the command name (in addition to the standard output
channel). This is accomplished with a new class OutputChannels in
commandio.h (and Output = OutputChannels&).

So far, this only affects the command system, but not the IPC interface,
which thus simply concatenates the two output channels in the reply to
'herbstclient'.

Thus, we can now print error messages to this dedicated channel and have
the error message prefixed with the command name automatically. Most of
the diff consists of migrating the error message printing to the new
perror(), which does the prefixing.

As a bonus, CallOrComplete does not need the command name anymore.

Also, HlwmCommon::callCommand() was only used in XMainLoop, so I have
used it there.

### CODE REVIEW

File: src/commandio.h [hunk 1]
```diff
@@ -69,7 +120,6 @@ class Input : public ArgList {
 class CallOrComplete {
 public:
     //! the name of the command that is called.
```
Comment (blarz) [inline src/commandio.h:122]: Can be removed now?

File: src/metacommands.cpp [hunk 2]
```diff
@@ -801,8 +801,12 @@ int MetaCommands::tryCommand(Input input, Output output) {
 
 int MetaCommands::silentCommand(Input input, Output output) {
     stringstream dummyOutput;
+    // discard output and error channel
+    // TODO: pass through the error channel as soon as
+    // the ipc-protocoll supports it.
```
Comment (blarz) [inline src/metacommands.cpp:806]: protocol

**Review by blarz — CHANGES_REQUESTED** [review]
Two minor findings, other than that, this looks good and grep shows me that you spotted all .command() occurrences

File: src/commandio.h (hunk 1 above)
Comment (t-wissmann) [reply src/commandio.h]: Indeed, thanks!

File: src/metacommands.cpp (hunk 2 above)
Comment (t-wissmann) [reply src/metacommands.cpp]: Fixed, thanks!

**Comment by t-wissmann** [comment]
> Two minor findings, other than that, this looks good and grep shows me that you spotted all .command() occurrences

Thanks, I've fixed them. Regarding `command()`, I had temporarily renamed `Input::command()` and `CallOrCommand::command()` and went through all compile errors :)

File: src/commandio.h (hunk 1 above)
Comment (blarz) [reply src/commandio.h]: Thanks!

**Review by blarz — APPROVED** [review]
Looks good to me!

**Pull request merged**